File: scripts/heatmap.py
# The puprose of that program is to show calendar hatmap of chosen feature form warsaw_data.csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import calendar
df = pd.read_csv('trial10/warsaw_data.csv')
df['date'] = pd.to_datetime(df['date'])
df['Year'] = df['date'].dt.year
df['Month'] = df['date'].dt.month
df['Day'] = df['date'].dt.day
pivot_df = df.pivot_table(values='pm2_5', index='Day', columns='Month', aggfunc=np.mean)
plt.figure(figsize=(10,10))
sns.heatmap(pivot_df, cmap='YlGnBu')
plt.title('Calendar Heatmap of PM2.5 values in 2022')
plt.figure(figsize=(10,10))
sns.heatmap(pivot_df, cmap='YlGnBu')
plt.title('Calendar Heatmap of PM2.5 values in 2022')
plt.savefig('heatmap.png')
plt.show()

File: scripts/heatmap.m
% Szymon Bartoszewicz
% The purpose of this script is to print heat xalendar for PM2.5.
data = readtable('warsaw_data.csv', 'VariableNamingRule', 'preserve');

data.date = datetime(data.date, 'InputFormat', 'yyyy-MM-dd');

data.Year = year(data.date);
data.Month = month(data.date);
data.Day = day(data.date);

% Calculate the mean PM2.5 values for each day of each month
meanPM25 = grpstats(data, {'Month', 'Day'}, 'mean', 'DataVars', 'pm2_5');

colorLimits = [min(meanPM25{:, 'mean_pm2_5'}(:)), max(meanPM25{:, 'mean_pm2_5'}(:))];

meanPM25 = unstack(meanPM25, 'mean_pm2_5', 'Month');

% Remove the first two columns
meanPM25(:, 1:2) = [];

figure;
h = heatmap(meanPM25.Variables, 'Colormap', parula);
h.Title = 'Calendar Heatmap of PM2.5 values in 2022';
h.XLabel = 'Month';
h.YLabel = 'Day';

h.ColorLimits = colorLimits;

% Display cell values with maximum 2 decimal places
h.CellLabelFormat = '%.2f';
